components/admin_console/permission_schemes_settings/permission_description.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+/* eslint-disable react/no-string-refs */
+
+import React, {useState, useRef, MouseEvent} from 'react';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+import {Overlay} from 'react-bootstrap';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import Tooltip from 'components/tooltip';
+
+import {generateId} from 'utils/utils';
+
+import {Role} from '@mattermost/types/roles';
+
+import {AdditionalValues} from './permissions_tree/types';
+
+type Props = {
+    intl: IntlShape;
+    id: string;
+    rowType: string;
+    inherited?: Partial<Role>;
+    selectRow: (id: string) => void;
+    additionalValues?: AdditionalValues | AdditionalValues['edit_post'];
+}
+
+const PermissionDescription = (props: Props) => {
+    const [open, setOpen] = useState(false);
+    const randomId = generateId();
+    const contentRef = useRef<HTMLSpanElement>(null);
+
+    const closeTooltip = () => setOpen(false);
+
+    const openTooltip = (e: MouseEvent) => {
+        const elm = e.currentTarget.querySelector('span');
+        const isElipsis = elm ? elm.offsetWidth < elm.scrollWidth : false;
+        setOpen(isElipsis);
+    };
+
+    const parentPermissionClicked = (e: MouseEvent) => {
+        const parent = (e.target as HTMLSpanElement).parentElement;
```

Comment:
you can avoid typecasting here if you specify the element type of the event itself 
```(e: React.MouseEvent<HTMLSpanElement>) ```

Reply:
Unfortunately didnt work I still receive `Property 'parentElement' does not exist on type 'EventTarget'`